### PR TITLE
Resolves  `eslint-plugin-jsdoc` test failures 

### PR DESCRIPTION
**Pull request by domlet**
- Adds new `eslint-plugin-jsdoc` tests:
```
check-line-alignment
require-returns-type
require-description
check-access
require-property
```
- Edits API docs to pass ^ these new tests.
- Makes a small number rive-by copyedits in API docs.

## Launch Checklist

 - [x] briefly describe the changes in this PR
 - [x] apply changelog label ('bug', 'feature', 'docs', etc) or use the label 'skip changelog'

### CODE REVIEW

**Review by HeyStenson — APPROVED** [review]
These changes look good to me @domlet! If we can get a 👍 from someone on the product team to double-check the `@returns` and other big changes that'd be nice ❤️

File: src/ui/camera.js [hunk 1]
```diff
@@ -162,7 +162,7 @@ class Camera extends Evented {
      * Returns the map's geographical centerpoint.
      *
      * @memberof Map#
-     * @returns The map's geographical centerpoint.
+     * @returns {LngLatLike} The map's geographical centerpoint.
```
Comment (SnailBones) [inline src/ui/camera.js:165]:
```suggestion
     * @returns {LngLat} The map's geographical centerpoint.
```
[LngLatLike](https://docs.mapbox.com/mapbox-gl-js/api/geography/#lnglatlike) is a flexible object for inputs. I presume the return value would be a LngLat.

Comment (SnailBones) [reply src/ui/camera.js:165]: Noticing that we also have: `@property {LngLatLike} center The desired center.` So maybe I'm wrong about this.

Comment (domlet) [reply src/ui/camera.js:165]: I see the example below says `return a LngLat object such as {lng: 0, lat: 0}` 
so I think you are correct here! @SnailBones

**Review by SnailBones — APPROVED** [review]
👍 Looks great to me

**Pull request merged**